src/data/nat/prime.lean

```diff
@@ -1102,41 +1102,63 @@ end nat
 namespace nat
 
 /-- The only prime divisor of positive prime power `p^k` is `p` itself -/
-lemma prime_pow_prime_divisor {p k : ℕ} (hk : 0 < k) (hp: prime p) :
+lemma prime_pow_prime_divisor {p k : ℕ} (hk : k ≠ 0) (hp : prime p) :
   (p^k).factors.to_finset = {p} :=
-by rw [hp.factors_pow, list.to_finset_repeat_of_ne_zero hk.ne']
+by rw [hp.factors_pow, list.to_finset_repeat_of_ne_zero hk]
 
-lemma mem_factors_mul_of_pos {a b : ℕ} (ha : 0 < a) (hb : 0 < b) (p : ℕ) :
+lemma mem_factors_mul {a b : ℕ} (ha : a ≠ 0) (hb : b ≠ 0) (p : ℕ) :
   p ∈ (a * b).factors ↔ p ∈ a.factors ∨ p ∈ b.factors :=
 begin
-  rw [mem_factors (mul_pos ha hb), mem_factors ha, mem_factors hb, ←and_or_distrib_left],
+  rw [mem_factors (mul_ne_zero ha hb).bot_lt, mem_factors ha.bot_lt, mem_factors hb.bot_lt,
```

Comment:
It might be worth changing `mem_factors` to take a `≠ 0` argument as well, to avoid having to add `.bot_lt` here.  It might not be too much work since it looks like `mem_factors` is only used twice outside this file.

Reply:
True, I held off on this for now though because I didn't want this PR to become too big! But as you say it's a small enough change so it should be fine